apps/asap-server/src/utils/sync-google-event.ts

```diff
@@ -0,0 +1,74 @@
+import Joi from '@hapi/joi';
+import { EventStatus } from '@asap-hub/model';
+import { calendar_v3 as calendarV3 } from 'googleapis';
+import { EventController } from '../controllers/events';
+import logger from './logger';
+
+export const syncEventFactory = (
+  eventsController: EventController,
+  calendarId: string,
+): ((
+  event: calendarV3.Schema$Event,
+  defaultTimezone: string,
+) => Promise<void>) => {
+  const syncEvent = async (
+    event: calendarV3.Schema$Event,
+    defaultTimezone: string,
+  ): Promise<void> => {
+    const schema = Joi.object({
+      id: Joi.string().required(),
+      summary: Joi.string().required(),
+      description: Joi.string(),
+      status: Joi.string().required(),
+      start: Joi.object({
+        date: Joi.string(),
+        dateTime: Joi.string(),
+        timeZone: Joi.string(),
+      }).or('date', 'dateTime'),
+      end: Joi.object({
+        date: Joi.string(),
+        dateTime: Joi.string(),
+        timeZone: Joi.string(),
+      }).or('date', 'dateTime'),
+    }).unknown();
+
+    const { error, value } = schema.validate(event);
+
+    if (error) {
+      logger('Ignored event update, validation error:', error);
+      return Promise.resolve();
```

Comment:
i though you were catching errors around syncEvent here: https://github.com/yldio/asap-hub/blob/cc96526794046211fc91fee8b59d92326b535974/apps/asap-server/src/utils/sync-google-calendar.ts#L47

so why not just throw when a validation error occurs?